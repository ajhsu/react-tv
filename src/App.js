import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

const videos = [
  'https://www.youtube.com/embed/2J5GzHoKl1Q',
  'https://www.youtube.com/embed/_xTGsIHRIdA',
  'https://www.youtube.com/embed/GF60Iuh643I',
  'https://www.youtube.com/embed/VAZ5te90vlc',
  'https://www.youtube.com/embed/j-nlYxx-1Ck'
];

class App extends Component {
  constructor() {
    super();
    this.state = {
      currentVideoIndex: 0,
      currentVideoSource: videos[0]
    };
  }
  componentDidMount() {
    // Auto-switching
    setInterval(() => {
      const nextIndex = this.state.currentVideoIndex + 1;
      this.setState({
        currentVideoIndex: nextIndex,
        currentVideoSource: videos[nextIndex]
      });
    }, 30 * 1000);
  }
  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>React TV</h2>
        </div>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <iframe
          width="640"
          height="360"
          src={this.state.currentVideoSource + '?autoplay=1'}
          frameBorder="0"
          allowFullScreen
        />
      </div>
    );
  }
}

export default App;
