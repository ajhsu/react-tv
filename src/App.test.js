import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';

it('renders without crashing', () => {
  // Smoke test: verifying that a component renders without throwing
  {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  }
  // Shallow rendering
  {
    const app = shallow(<App />);
    expect(app).toMatchSnapshot();
  }
});
