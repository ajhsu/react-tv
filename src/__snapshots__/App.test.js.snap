// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="App"
>
    <div
        className="App-header"
    >
        <img
            alt="logo"
            className="App-logo"
            src="logo.svg"
        />
        <h2>
            React TV
        </h2>
    </div>
    <p
        className="App-intro"
    >
        To get started, edit 
        <code>
            src/App.js
        </code>
         and save to reload.
    </p>
    <iframe
        allowFullScreen={true}
        frameBorder="0"
        height="360"
        src="https://www.youtube.com/embed/2J5GzHoKl1Q?autoplay=1"
        width="640"
    />
</div>,
  "nodes": Array [
    <div
      className="App"
>
      <div
            className="App-header"
      >
            <img
                  alt="logo"
                  className="App-logo"
                  src="logo.svg"
            />
            <h2>
                  React TV
            </h2>
      </div>
      <p
            className="App-intro"
      >
            To get started, edit 
            <code>
                  src/App.js
            </code>
             and save to reload.
      </p>
      <iframe
            allowFullScreen={true}
            frameBorder="0"
            height="360"
            src="https://www.youtube.com/embed/2J5GzHoKl1Q?autoplay=1"
            width="640"
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 11,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "currentVideoIndex": 0,
          "currentVideoSource": "https://www.youtube.com/embed/2J5GzHoKl1Q",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="App"
>
          <div
                    className="App-header"
          >
                    <img
                              alt="logo"
                              className="App-logo"
                              src="logo.svg"
                    />
                    <h2>
                              React TV
                    </h2>
          </div>
          <p
                    className="App-intro"
          >
                    To get started, edit 
                    <code>
                              src/App.js
                    </code>
                     and save to reload.
          </p>
          <iframe
                    allowFullScreen={true}
                    frameBorder="0"
                    height="360"
                    src="https://www.youtube.com/embed/2J5GzHoKl1Q?autoplay=1"
                    width="640"
          />
</div>,
        "_debugID": 12,
        "_renderedOutput": <div
          className="App"
>
          <div
                    className="App-header"
          >
                    <img
                              alt="logo"
                              className="App-logo"
                              src="logo.svg"
                    />
                    <h2>
                              React TV
                    </h2>
          </div>
          <p
                    className="App-intro"
          >
                    To get started, edit 
                    <code>
                              src/App.js
                    </code>
                     and save to reload.
          </p>
          <iframe
                    allowFullScreen={true}
                    frameBorder="0"
                    height="360"
                    src="https://www.youtube.com/embed/2J5GzHoKl1Q?autoplay=1"
                    width="640"
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
